mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -1064,9 +1068,15 @@ fn try_parse_generic_error(request_result: &MetaRequestResult) -> Option<S3Reque
         // redirect
         400 => try_parse_forbidden(request_result).or_else(|| try_parse_redirect(request_result)),
         403 => try_parse_forbidden(request_result),
-        0 => try_parse_throttled(request_result).or_else(|| {
-            try_parse_canceled_request(request_result).or_else(|| try_parse_no_credentials(request_result))
-        }),
+        0 => {
+            if let Some(err) = try_parse_throttled(request_result) {
```

Comment:
nit: this is fine, but inconsistent with the `400` case. I'd leave it as a chain of `.or_else`.

Reply:
do you mind adding a suggestion of how you'd expect it to look like? :)

(I though the current version is what you've mentioned previously by "flatten": https://github.com/awslabs/mountpoint-s3/pull/882#discussion_r1633067922)

Reply:
will something like this be better?
```rust
        0 => try_parse_throttled(request_result)
            .or_else(|| try_parse_canceled_request(request_result))
            .or_else(|| Some(try_parse_no_credentials_or_generic(request_result))),
```

Reply:
will keep this variant in new commit, looks neat

Reply:
Yes, I meant the latter variant. "flatten" compared to nesting the second `or_else` inside the first. Sorry I wasn't clear :)